File: Search_BinarySearch.cpp
// Binary Search Algorithm

#include <bits/stdc++.h>
using namespace std;
int binarySearch(int arr[], int size, int target) {
    int left = 0;
    int right = size - 1;
    while(left <= right) {
        int mid = (left + right) / 2;
        if(arr[mid] == target) return mid;
        if(arr[mid] < target) left = mid + 1;
        else right = mid - 1;
    }
    return -1;
}

int main() {

    int arr[] = {1, 2, 3, 4, 5, 6, 7 };
    int target = 4;
    int size = sizeof(arr) / sizeof(arr[0]);
    int result = binarySearch(arr, size, target);
    if(result != -1) {
        cout << target << " found at index " << result << endl;
    } else {
        cout << target << " not found" << endl;
    }

    return 0;
}

File: Algorithm/Searchin Algorithm/Search_BinarySearch.py
# Binary Search

def binarySearch(arr, item):
    left = 0
    right = len(arr) - 1
    while(left <= right):
        mid = (left + right) // 2
        if arr[mid] == item:
            return mid
        if arr[mid] < item:
            left = mid + 1
        else:
            right = mid - 1
    else:
        return False

arr = [1, 2, 4, 5, 6, 7, 8, 9, 55]
item = 9

if(binarySearch(arr, item)):
    print(item, 'Fount at index', binarySearch(arr, item))
else:
    print(item, 'Not found')